### PR TITLE
fix updating CSS on push_screen

### PR DESCRIPTION
**Pull request by willmcgugan**
Fixes https://github.com/Textualize/textual/issues/3217
Fixes https://github.com/Textualize/textual/issues/3221

We recently added `CSS` to Screens, which requires a `stylesheet.update` when pushing the screen.

### CODE REVIEW

**Comment by dmunozv04** [comment]
@willmcgugan This fix seems to not work correctly, since the bug also appears without using CSS.
image

File: src/textual/app.py [hunk 1]
```diff
@@ -2059,7 +2061,8 @@ async def invoke_ready_callback() -> None:
                     Reactive._initialize_object(self)
 
                     self.stylesheet.update(self)
-                    self.refresh()
+                    if self.screen is not default_screen:
+                        self.stylesheet.update(default_screen)
```
Comment (rodrigogiraoserrao) [inline src/textual/app.py:2065]: What can happen in between the time where you set `default_screen = self.screen` and this `if` statement that `self.screen` is no longer what you stored inside `default_screen`?

Comment (willmcgugan) [reply src/textual/app.py:2065]: The issue was that if you change the screen stack in App.on_mount then the default screen never has its CSS applied.

Comment (rodrigogiraoserrao) [reply src/textual/app.py:2065]: Ahhhh, I see. Thanks.

**Review by rodrigogiraoserrao — APPROVED** [review]

**Pull request merged**